v3io/dataplane/transport/httpclient.py

```diff
@@ -166,6 +167,11 @@ def _send_request_on_connection(self, request, connection):
                     connection=connection,
                 )
                 connection.close()
+                if isinstance(request.body, mmap.mmap):
```

Comment:
In general, request.body can be any seekable object (BytesIO for example, or file handle). Also it is not guaranteed to start with offset 0. So I would change it to something like:
before the request:

if hasattr(request.body, 'seekable') and request.body.seekable():
    offset = request.body.tell()
....

on failure seek back: 

request.body.seek(offset)